[CI] ParseIpTests testLeadingZerosAreOctal {TestCase[str=v4, validLeadingZerosRejected=true, validLeadingZerosAreDecimal=true, validLeadingZerosAreOctal=true]} failing

**Build Scans:**
- [elasticsearch-periodic #7523 / release-tests](https://gradle-enterprise.elastic.co/s/d73hpga2xkgpy)
- [elasticsearch-intake #21547 / part3](https://gradle-enterprise.elastic.co/s/l5cnaie4uhxy4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.scalar.convert.ParseIpTests.testLeadingZerosAreOctal {TestCase[str=v4, validLeadingZerosRejected=true, validLeadingZerosAreDecimal=true, validLeadingZerosAreOctal=true]}" -Dtests.seed=2CCF2C213AEEE770 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=nl-NL -Dtests.timezone=Antarctica/DumontDUrville -Druntime.java=24
```

**Applicable branches:**
9.0

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.scalar.convert.ParseIpTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testLeadingZerosAreOctal%20%7BTestCase%5Bstr%3Dv4%2C%20validLeadingZerosRejected%3Dtrue%2C%20validLeadingZerosAreDecimal%3Dtrue%2C%20validLeadingZerosAreOctal%3Dtrue%5D%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <[0 0 0 0 0 0 0 0 0 0 ff ff c3 0 5e 83]>
     but: was <[0 0 0 0 0 0 0 0 0 0 ff ff c3 0 88 cb]>
```

**Issue Reasons:**
- [9.0] 2 failures in test testLeadingZerosAreOctal {TestCase[str=v4, validLeadingZerosRejected=true, validLeadingZerosAreDecimal=true, validLeadingZerosAreOctal=true]} (10.5% fail rate in 19 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ParseIpTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ParseIpTests.java
@@ -191,6 +191,7 @@ public class ParseIpTests extends ESTestCase {
                 lastWasBreak = true;
                 b.append(current).append('.');
                 current = 0;
+                octalMode = false;
                 continue;
             }
             lastWasBreak = false;

